File: MyMonitorHub.Agent/MyMonitorHub.Agent.Core/Core/DOM.cs
using Newtonsoft.Json.Linq;

namespace MyMonitorHub.Agent.Core
{
    public class Dom
    {
        public static string? GetChildElementText(JToken node, string propertyName)
        {
            return node?[propertyName]?.ToString();
        }

        public static string GetChildElementText(JToken node, string propertyName, string deflt)
        {
            var value = node?[propertyName]?.ToString();
            return string.IsNullOrEmpty(value) ? deflt : value;
        }

        public static string? GetAttribute(JToken node, string attributeName, string? dflt = null)
        {
            var value = node?[attributeName]?.ToString();
            return value ?? dflt;
        }
    }
}

File: MyMonitorHub.JavaAgent/src/main/java/mymonitorhub/agent/core/Dom.java
package mymonitorhub.agent.core;

import com.fasterxml.jackson.databind.JsonNode;

/**
 * JSON attribute helper — mirrors C# Dom.GetAttribute.
 */
public final class Dom {

    private Dom() {
    }

    public static String getAttribute(JsonNode node, String name) {
        return getAttribute(node, name, null);
    }

    public static String getAttribute(JsonNode node, String name, String defaultValue) {
        if (node == null || node.isMissingNode() || node.isNull()) {
            return defaultValue;
        }
        JsonNode child = node.get(name);
        if (child == null || child.isNull() || child.isMissingNode()) {
            return defaultValue;
        }
        String val = child.asText(defaultValue);
        return (val == null || val.isEmpty()) ? defaultValue : val;
    }
}
